src/main/java/ro/msg/learning/shop/service/impl/OrdersServiceImpl.java

```diff
@@ -2,106 +2,32 @@
-        for (ProductQuantityPairDto productQuantityPairDto : productQuantityPairDtoList) {
-            Product product = getProductFromPair(productQuantityPairDto);
-            locationList.removeIf(location -> !isProductOnLocation(location, product, productQuantityPairDto.getQuantity()));
-        }
-        Optional<Location> optionalLocation = locationList.stream().findFirst();
-        return optionalLocation.orElse(null);
-    }
-
-    public void subtractStock(Location location, ProductQuantityPairDto productQuantityPairDto) {
-        Product product = getProductFromPair(productQuantityPairDto);
-        Stock stock = getStock(product, location);
-        int quantity = stock.getQuantity() - productQuantityPairDto.getQuantity();
-        Stock updatedStock = new Stock(product, location, quantity);
-        stockService.updateStock(updatedStock);
-    }
-
-    public void subtractAllStocks(Location location, List<ProductQuantityPairDto> productQuantityPairDtoList) {
-        for (ProductQuantityPairDto productQuantityPairDto : productQuantityPairDtoList) {
-            subtractStock(location, productQuantityPairDto);
-        }
-    }
-
-    public Set<OrderDetail> createOrderDetails(Orders order, Location location, List<ProductQuantityPairDto> productQuantityPairDtoList) {
-        Set<OrderDetail> orderDetails = new HashSet<>();
-        for (ProductQuantityPairDto productQuantityPairDto : productQuantityPairDtoList) {
-            Product product = getProductFromPair(productQuantityPairDto);
-            int quantity = productQuantityPairDto.getQuantity();
-            OrderDetail orderDetail = new OrderDetail(order, product, location, quantity);
-            orderDetailService.createOrderDetail(orderDetail);
-            orderDetails.add(orderDetail);
-        }
-        return orderDetails;
-    }
-
-    public Orders handleOrderCreation(Orders order, List<ProductQuantityPairDto> productQuantityPairDtoList) {
-        Customer customer = customerService.getCustomer(order.getCustomer().getId());
-        Location location = verifyStock(productQuantityPairDtoList);
-        if (location != null) {
-            order.setCustomer(customer);
-            Orders newOrder = createOrders(order);
-            Set<OrderDetail> orderDetails = createOrderDetails(newOrder, location, productQuantityPairDtoList);
-            subtractAllStocks(location, productQuantityPairDtoList);
-            Set<Orders> orders = customer.getOrdersList();
-            orders.add(order);
-            customer.setOrdersList(orders);
-            newOrder.setOrdersDetails(orderDetails);
-            newOrder.setCustomer(customer);
-            return ordersRepository.save(newOrder);
-        } else {
-            throw new OrderNotCreatedException("your order was not created!");
-        }
+    public OrdersDto handleOrderCreation(Orders order, Set<ProductQuantityPairUtil> productQuantityPairUtilList) {
```

Comment:
Don't return a DTO from the Service Layer. Your ProductQuantityPairUtil is actually an OrderDetail. The Order contains the order details. Please adapt your methods.